src/natcap/invest/sdr/sdr_core.pyx

```diff
@@ -419,49 +455,59 @@ def calculate_sediment_deposition(
                 seed_col = xoff + col_index
-                # search to see if this is a good seed
-                if mfd_flow_direction_raster.get(seed_col, seed_row) == 0:
+                # check if this is a good seed
+                # a good seed pixel is a local high point
+                if mfd_flow_direction_raster.get(seed_col, seed_row) == mfd_nodata:
                     continue
                 seed_pixel = 1
+                # iterate over each of the pixel's neighbors
                 for j in range(8):
+                    # skip if the neighbor is outside the raster bounds
                     neighbor_row = seed_row + ROW_OFFSETS[j]
                     if neighbor_row < 0 or neighbor_row >= n_rows:
                         continue
                     neighbor_col = seed_col + COL_OFFSETS[j]
                     if neighbor_col < 0 or neighbor_col >= n_cols:
                         continue
+                    # skip if the neighbor's flow direction is undefined
                     neighbor_flow_val = <int>mfd_flow_direction_raster.get(
                         neighbor_col, neighbor_row)
-                    if neighbor_flow_val == 0:
+                    if neighbor_flow_val == mfd_nodata:
                         continue
+                    # if the neighbor flows into it, it's not a local high
+                    # point and so can't be a seed pixel
                     neighbor_flow_weight = (
                         neighbor_flow_val >> (inflow_offsets[j]*4)) & 0xF
                     if neighbor_flow_weight > 0:
-                        # neighbor flows in, not a seed
-                        seed_pixel = 0
+                        seed_pixel = 0  # neighbor flows in, not a seed
                         break
-                if seed_pixel and (
-                        sediment_deposition_raster.get(
-                            seed_col, seed_row) ==
-                        sediment_deposition_nodata):
+
+                # if this can be a seed pixel and hasn't already been
+                # calculated, put it on the stack
+                if seed_pixel and sediment_deposition_raster.get(
+                        seed_col, seed_row) == target_nodata:
                     processing_stack.push(seed_row * n_cols + seed_col)
 
                 while processing_stack.size() > 0:
                     # loop invariant: cell has all upstream neighbors
-                    # processed
+                    # processed. this is true for seed pixels because they
+                    # have no upstream neighbors.
                     flat_index = processing_stack.top()
                     processing_stack.pop()
                     global_row = flat_index // n_cols
                     global_col = flat_index % n_cols
 
-                    # calculate the upstream Fj contribution to this pixel
+                    # (sum over j ∈ J of f_j * p(i,j) in the equation for r_i)
```

Comment:
The `element of` epsilon symbol doesn't show up in my `vim` text editor! I guess I need to update some vim plugins.

Reply:
Oh weird! I like being able to use unicode but I have also run into a few more obscure symbols that don't render.

Reply:
@dcdenu4 are you using a unicode-compatible terminal for your vim session?  I think a modern vim/neovim should be able to handle UTF-8, but only if your terminal does too!

Reply:
I'm using Cmder. I'll have to check!